### PR TITLE
Skills Decorations

### PR DESCRIPTION
**Pull request by Linkaaaaa**
# Chronomancer
- Well of Eternity
- Well of Action
- Well of Calamity
- Well of Precognition
- Well of Senility
- Gravity well

# Specter
- Well of Gloom
- Well of Bounty
- Well of Tears
- Well of Silance
- Well of Sorrow
- Shadowfall

# Harbinger
- Vital Draw

# Necromancer
- Mark of Blood or Chillblains (Mark)
- Mark of Blood (Active)
- Chillblains (Active)
- Putrid Mark (Mark and Active)
- Reaper's Mark (Mark and Active)

[test_logs.zip](https://github.com/baaron4/GW2-Elite-Insights-Parser/files/12910172/test_logs.zip)

### CODE REVIEW

File: GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs [hunk 1]
```diff
@@ -84,5 +86,173 @@ internal static bool IsKnownMinionID(int id)
...
+                    (int, int) lifespan = ProfHelper.ComputeEffectLifespan(log, effect, 5000);
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, 0, 240, lifespan, color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                    replay.Decorations.Add(new IconDecoration(ParserIcons.EffectWellOfAction, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector).UsingSkillMode(skill));
+                    
+                    // Well pulses - Hard coded because the effects don't have a Src
+                    int pulseTimeDelay = 0;
+                    for (int i = 0; i < 4; i++)
+                    {
+                        int effectTimeStart = (int)effect.Time + pulseTimeDelay;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs:133]: As this is hardcoded, the for loop should break if effectTimeStart > lifespan.item2.

File: GW2EIEvtcParser/EIData/ProfHelpers/Necromancer/HarbingerHelper.cs [hunk 2]
```diff
@@ -48,5 +51,22 @@ public static bool IsHarbingerShroudTransform(long id)
...
         }
 
+        internal static void ComputeProfessionCombatReplayActors(AbstractPlayer player, ParsedEvtcLog log, CombatReplay replay)
+        {
+            Color color = Colors.Necromancer;
+
+            // Vital Draw
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.HarbingerVitalDrawAoE, out IReadOnlyList<EffectEvent> vitalDraws))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Chronomancer, VitalDraw);
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/Necromancer/HarbingerHelper.cs:61]: Spec.Harbinger

File: GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs [hunk 3]
```diff
@@ -84,5 +86,175 @@ internal static bool IsKnownMinionID(int id)
...
+                    (int, int) lifespan = ProfHelper.ComputeEffectLifespan(log, effect, 5000);
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, 0, 240, lifespan, color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                    replay.Decorations.Add(new IconDecoration(ParserIcons.EffectWellOfSenility, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector).UsingSkillMode(skill));
+
+                    // Well pulses - Hard coded because the effects don't have a Src
+                    int pulseTimeDelay = 0;
+                    for (int i = 0; i < 4; i++)
+                    {
+                        int effectTimeStart = (int)effect.Time + pulseTimeDelay;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs:216]: Same situation as Well of Action

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs:216]: you mean if (effectTimeStart > lifespan.Item2) { break; } ? I have already added it

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**